doc/src/guides/solving/solve-equation-algebraically.md

```diff
@@ -0,0 +1,413 @@
+{-sqrt(y), sqrt(y)}
+```
+
+Here are recommendations on when to use:
+
+- {func}`~.solve`
+    - You want to get explicit symbolic representations of the different 
+    values a variable could take that would satisfy the equation.
+    - You want to substitute those explicit solution values into other 
+    equations or expressions involving the same variable using 
+    {meth}`~sympy.core.basic.Basic.subs`
+
+- {func}`~.solveset`
+    - You want to represent the solutions in a mathematically precise way, 
+    using [mathematical sets](../../modules/sets.rst).
+    - You want a representation of all the solutions, including if there are 
+    infinitely many.
+    - You want a consistent input interface.
+    - You want to limit the domain of the solutions to any arbitrary set.
+    - You do not need to programmatically extract solutions from the solution 
+    set: solution sets cannot necessarily be interrogated programmatically.
+
+## Guidance
+
+### Include the variable to be solved for in the function call
+
+We recommend you include the variable to be solved for as the second argument 
+for either function. While this is optional for equations with a single 
+symbol, it is a good practice because it ensures 
+SymPy will solve for the desired symbol. For example, you may expect the 
+following to solve for $x$, 
+and SymPy will solve for $y$:
+
+```py
+>>> from sympy.abc import x, y
+>>> from sympy import solve
+>>> solve(x ** 2 - y, dict=True)
+[{y: x**2}]
+```
+
+Specifying the variable to solve for ensures that SymPy solves for it:
+
+```py
+>>> from sympy.abc import x, y
+>>> from sympy import solve
+>>> solve(x ** 2 - y, x, dict=True)
+[{x: -sqrt(y)}, {x: sqrt(y)}]
+```
+
+### Ensure consistent formatting from {func}`~.solve` by using `dict=True`
+
+{func}`~.solve` produces various output formats depending on the answer, 
+unless you use `dict=True` to ensure the result will be formatted as a 
+dictionary. We recommend using `dict=True`, especially if you want to 
+extract information from the result programmatically.
+
+## Solve an equation using {func}`~.solve` or {func}`~.solveset`
+
+You can solve an equation in several ways. 
+The examples below demonstrate using both {func}`~.solve` and 
+{func}`~.solveset` where applicable. 
+You can choose the function best suited to your equation.
+
+### Make your equation into an expression that equals zero
+
+Use the fact that any expression not in an `Eq` (equation) is automatically 
+assumed to equal zero (0) by the solving functions. You can rearrange the 
+equation $x^2 = y$ to $x^2 - y = 0$, and solve that expression. This approach 
+is convenient if you are interactively solving an expression which already 
+equals zero, or an equation that you do not mind rearranging to 
+$expression = 0$.
+
+```py
+>>> from sympy import solve, solveset
+>>> from sympy.abc import x, y
+>>> solve(x ** 2 - y, x, dict=True)
+[{x: -sqrt(y)}, {x: sqrt(y)}]
+>>> solveset(x**2 - y, x)
+{-sqrt(y), sqrt(y)}
```

Comment:
In many places code snippets are inconsistent in the spacing around `**`. I prefer no spaces like `x**2` but either way it should be consistent.

Reply:
Removed spaces around `**` so is like `x**2` in #23805